src/pages/correspondent/Forms/CorrespondentForm.js

```diff
@@ -68,140 +60,131 @@ const CorrespondentForm = ({
-          required
-          maxLength='50'
-          maxAccess={maxAccess}
-          onChange={formik.handleChange}
-          onClear={() => formik.setFieldValue('name', '')}
-          error={formik.touched.name && Boolean(formik.errors.name)}
-          helperText={formik.touched.name && formik.errors.name}
-        />
-      </Grid>
-
-         <Grid item xs={12}>
-        <ResourceLookup
-         endpointId={BusinessPartnerRepository.MasterData.snapshot}
-          name='bpRef'
-          required
-          label={labels.businessPartner}
-          valueField='reference'
-          displayField='name'
-          valueShow='bpRef'
-          secondValueShow='bpName'
-          form={formik}
-          onChange={(event, newValue) => {
-            if (newValue) {
-              formik.setFieldValue('bpId', newValue?.recordId)
-              formik.setFieldValue('bpRef', newValue?.reference)
-              formik.setFieldValue('bpName', newValue?.name)
-            } else {
-              formik.setFieldValue('bpId', '')
-              formik.setFieldValue('bpRef', '')
-              formik.setFieldValue('bpName', '')
+  return (
+    <FormShell form={formik} resourceId={ResourceIds.Correspondent} maxAccess={maxAccess} editMode={editMode}>
+      <Grid container spacing={4}>
+        <Grid item xs={12}>
+          <CustomTextField
+            name='reference'
+            label={labels.reference}
+            value={formik.values.reference}
+            required
+            onChange={formik.handleChange}
+            maxLength='10'
+            maxAccess={maxAccess}
+            onClear={() => formik.setFieldValue('reference', '')}
+            error={formik.touched.reference && Boolean(formik.errors.reference)}
+            helperText={formik.touched.reference && formik.errors.reference}
+          />
+        </Grid>
+        <Grid item xs={12}>
+          <CustomTextField
+            name='name'
+            label={labels.name}
+            value={formik.values.name}
+            required
+            maxLength='50'
+            maxAccess={maxAccess}
+            onChange={formik.handleChange}
+            onClear={() => formik.setFieldValue('name', '')}
+            error={formik.touched.name && Boolean(formik.errors.name)}
+            helperText={formik.touched.name && formik.errors.name}
+          />
+        </Grid>
+
+        <Grid item xs={12}>
+          <ResourceLookup
+            endpointId={BusinessPartnerRepository.MasterData.snapshot}
+            name='bpRef'
+            required
+            label={labels.businessPartner}
+            valueField='reference'
+            displayField='name'
+            valueShow='bpRef'
+            secondValueShow='bpName'
+            form={formik}
+            onChange={(event, newValue) => {
+              if (newValue) {
+                formik.setFieldValue('bpId', newValue?.recordId)
+                formik.setFieldValue('bpRef', newValue?.reference)
```

Comment:
can we substitute this -if else- blocks to ternary conditional operator (for less code)